mreg/models.py

```diff
@@ -574,43 +575,43 @@ def unused_addresses(self):
         Returns which ipaddresses on the network are unused.
         """
         network_ips = []
-        unusable = self.get_unusable_ipaddresses()
-        not_available = unusable | self.used_addresses
-        if self.network.num_addresses > MAX_UNUSED_LIST:
-            # Getting all availible IPs for a ipv6 prefix can easily cause
-            # the webserver to hang due to lots and lots of IPs. Instead limit
-            # to the first MAX_UNUSED_LIST hosts.
-            found = 0
-            for ip in self.network.hosts():
-                if ip in not_available:
-                    continue
-                network_ips.append(ip)
-                found += 1
-                if found == MAX_UNUSED_LIST:
-                    break
-            return set(network_ips)
-        else:
-            return set(self.network.hosts()) - not_available
+        used = self.used_addresses
+        # Getting all availible IPs for a ipv6 prefix can easily cause
+        # the webserver to hang due to lots and lots of IPs. Instead limit
+        # to the first MAX_UNUSED_LIST hosts.
+        found = 0
+        for ip in self.network.hosts():
+            if ip in used or not self.is_usable(ip):
+                continue
```

Comment:
This will break in a similar way as previously if e.g. the exclude range is from `2001:db8::` to `2001:db8::ffff:0:0` just a tiny fraction on the `2001:db8::/64` network, but would still take time to iterate 2^48 addresses.

Maybe have some logic to check for first valid ip after an exclude range, instead of blindly attempting an ipaddress and then the next etc.